### PR TITLE
2.0 preview

### PR DESCRIPTION
**Pull request by valdrinkoshi**
Consists in keeping the old implementation only if we are in Polymer 1.x, otherwise have an empty implementation for `attached/detached` since `iron-form#2.x` doesn't rely on this behavior.

### CODE REVIEW

**Comment by notwaldorf** [comment]
This is some new sass, googlebot.

File: README.md [hunk 1]
```diff
@@ -21,5 +21,5 @@ _[Demo and API docs](https://elements.polymer-project.org/elements/iron-form-ele
 
   Polymer.IronFormElementBehavior enables a custom element to be included
   in an `iron-form`.
-
-
+  If your form element is used with Polymer 2.0, it doesn't need to implement
+  `Polymer.IronFormElementBehavior` as `iron-form` won't rely on it.
```
Comment (notwaldorf) [inline README.md]: I would add `iron-form#2.0.0` won't rely on it, to be specific

File: test/basic.html [hunk 2]
```diff
@@ -29,6 +29,15 @@
   </test-fixture>
 
   <script>
+HTMLImports.whenReady(function() {
```
Comment (notwaldorf) [inline test/basic.html]: Maybe `WebComponentsReady` instead? also, indentation is weird

File: README.md (hunk 1 above)
Comment (valdrinkoshi) [reply README.md]: updated 👌

File: test/basic.html (hunk 2 above)
Comment (valdrinkoshi) [reply test/basic.html]: true, done!

File: demo/index.html [hunk 3]
```diff
@@ -50,6 +50,10 @@
 
     <script>
       function update() {
+        if (Polymer.version[0] !== '1') {
```
Comment (notwaldorf) [inline demo/index.html]: I think we're using `if (Polymer.Element)` everywhere

File: iron-form-element-behavior.html [hunk 4]
```diff
@@ -15,10 +15,12 @@
   Polymer.IronFormElementBehavior enables a custom element to be included
   in an `iron-form`.
 
+  Empty implementation for Polymer 2.0
+
   @demo demo/index.html
   @polymerBehavior
   */
-  Polymer.IronFormElementBehavior = {
+  Polymer.IronFormElementBehavior = Polymer.version[0] !== '1' ? {} : {
```
Comment (notwaldorf) [inline iron-form-element-behavior.html]: Same comment as above

File: test/basic.html [hunk 5]
```diff
@@ -29,43 +31,54 @@
...
 
   <script>
-    suite('basic', function() {
-      var form;
+    window.addEventListener('WebComponentsReady', function() {
 
-      setup(function() {
-        form = fixture('basic');
-      });
+      if (Polymer.version[0] !== '1') {
```
Comment (notwaldorf) [inline test/basic.html]: Polymer.Element

**Review by notwaldorf — CHANGES_REQUESTED** [review]

**Comment by bicknellr** [comment]
IronCheckedElementBehavior's `_getValidity` method (used by IronValidatableBehavior) relies on `required`, which was previously provided by this behavior. What's the intended upgrade path? Should I add `required` to IronCheckedElementBehavior?

**Comment by bicknellr** [comment]
Maybe IronFormElementBehavior should still provide properties and just remove the `attached` / `detached` for 2.x?

**Comment by notwaldorf** [comment]
ping @valdrinkoshi for the above

**Comment by valdrinkoshi** [comment]
@bicknellr @notwaldorf done!

File: test/basic.html [hunk 6]
```diff
@@ -29,43 +31,48 @@
   </test-fixture>
 
   <script>
-    suite('basic', function() {
-      var form;
+    window.addEventListener('WebComponentsReady', function() {
```
Comment (notwaldorf) [inline test/basic.html]: Why is this needed? You already have a `Polymer.base.async` for distribution

Comment (valdrinkoshi) [reply test/basic.html]: oh right! this was needed before since we were skipping the whole test suite instantiation (we had to wait for Polymer to be defined)
This is not needed anymore, removing it now!

Comment (valdrinkoshi) [reply test/basic.html]: done with https://github.com/PolymerElements/iron-form-element-behavior/pull/30/commits/73df16b5f5d95ddd4aa8bf9abb492fb3ca48bbc4

**Comment by notwaldorf** [comment]
@valdrinkoshi travis says your bowers are grumpy :/

**Comment by valdrinkoshi** [comment]
@notwaldorf tes were broken because of `iron-flex-layout#2.0-preview` bower.json which just got fixed. Tests should be 💚  in a few moments

File: iron-form-element-behavior.html [hunk 7]
```diff
@@ -32,7 +34,7 @@
        *
        * @event iron-form-element-unregister
        */
-
+
```
Comment (bicknellr) [inline iron-form-element-behavior.html:37]: extra spaces

**Comment by bicknellr** [comment]
Thanks for adding those back, IronCheckedElementBehavior's demos are working again!

**Review by notwaldorf — APPROVED** [review]
My name is Meownica and I approve of this element 👍
